src/ai/backend/manager/models/vfolder.py

```diff
@@ -115,9 +115,10 @@ class VFolderOperationStatus(str, enum.Enum):
     """
 
     READY = "ready"
-    PREPARED = "prepared"
+    PERFORMING = "performing"
     CLONING = "cloning"
     DELETING = "deleting"
+    DELETED = "deleted"
     MOUNTED = "mounted"
```

Comment:
@fregataa I can't define clear approach about how to prevent clone/delete operation when the target vfolder is being mounted to one or more sessions. Can I get some hints?

Reply:
I guess we can change the status in `manager.api.vfolder.clone()` to `CLONING` and restore it in `_clone_bgtask()` method. then the status will change after background task finishes.
but I am not sure whether we should block the access or not.
you can try to apply it and let reviewers consider of it.

Reply:
@fregataa Does simply it mean to restore the vfolder status update query in `clone_bgtask()` method, so the status can be updated right after the background task is completed?

```python
async def _clone_bgtask(reporter: ProgressReporter) -> None:
    async with root_ctx.storage_manager.request(
        source_folder_host,
        "POST",
        "folder/clone",
        json={
            "src_volume": source_volume_name,
            "src_vfid": str(row["id"]),
            "dst_volume": target_volume_name,
            "dst_vfid": str(folder_id),
        },
    ):
        pass
    async with root_ctx.db.begin() as conn:
        query = (
            sa.update(vfolders)
            .values(status=VFolderOperationStatus.READY)
            .where(vfolders.c.name == row["name"])
        )
        await conn.execute(query)   

task_id = await root_ctx.background_task_manager.start(_clone_bgtask)
```

Reply:
yes. and you can relocate the db query context block into request context